Cubical/Structures/Group.agda

```diff
@@ -0,0 +1,171 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Structures.Group where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.HLevels
+open import Cubical.Data.Sigma
+
+open import Cubical.Foundations.SIP renaming (SNS-PathP to SNS)
+
+open import Cubical.Structures.NAryOp
+open import Cubical.Structures.Semigroup hiding (⟨_⟩)
+open import Cubical.Structures.Monoid using (Monoids; inv-lemma)
+
+
+private
+  variable
+    ℓ ℓ' : Level
+
+raw-group-structure : Type ℓ → Type ℓ
+raw-group-structure = raw-semigroup-structure
+
+raw-group-is-SNS : SNS {ℓ} raw-group-structure _
+raw-group-is-SNS = raw-semigroup-is-SNS
+
+group-axioms : (G : Type ℓ) → raw-group-structure G → Type ℓ
```

Comment:
Why define groups like this? I think it's better to separate the operations from the axioms as it simplifies the proof that the axioms form a prop later?

Reply:
Well, the main reason is that when you apply SIP with groups with an inverse function (the same applies when you put the neutral element in the raw-structure, as we can see from the Monoid implementation) you get that an isomorphism is an equivalence that preserves multiplication and inversion, while group isomorphisms are defined as preserving only multiplication. I thought an alternative that was to define a second notion of isomorphism (as equivalences that only preserves multiplication) and prove that in fact they were the same (since it can be proven that a function that respect multiplication respects inversion). But I thought it was more difficult that this and actually ,i think more faithful to the mathematical tradition (where Groups are normally defined as having only a multiplication and the inversion and neutral element are defined from axioms).

Reply:
Ah I see, that makes sense. Can you add a comment about it before `group-axioms`?

Reply:
-- The neutral element and the inverse function will be derived from the
-- axioms, instead of being defined in the raw-group-structure in order
-- to have that isomorphisms between groups are equivalences that preserves
-- multiplication (so we don't have to show that they also preserve inversion
-- and neutral element, although they will preserve them).

Is this good enough?

Reply:
Sounds good!